src/Common/AbstractModel.m

```diff
@@ -407,6 +407,58 @@ function saveObj(obj, suffix)
         end
 
 
+    end
+
+    methods(Static)
+
+        function FitProvenance = getProvenance(varargin)
+            
+            FitProvenance = struct();
+            FitProvenance.EstimationSoftwareName = 'qMRLab';
+            FitProvenance.EstimationSoftwareVer  = qMRLabVer;
+
+            % Add extra fields
+            if nargin>0
+
+                if any(cellfun(@isequal,varargin,repmat({'extra'},size(varargin))))
+                    idx = find(cellfun(@isequal,varargin,repmat({'extra'},size(varargin)))==1);
+                    if isstruct(varargin{idx+1})
+
+                        tmp = varargin{idx+1};
+                        names = fieldnames(tmp);
+                        for ii=1:length(names) 
+                            FitProvenance.(names{ii}) = tmp.(names{ii});
+                        end
+
+                    end    
+                end
+
```

Comment:
any particular reason for those extra lines?

Reply:
Yes, resultant maps will contain metadata fields such as `EstimationReference`, to be provided by wrapper scripts as `extra` struct fields. Example use:

```
addField = struct();
addField.EstimationReference =  'Helms, G. et al. (2008), Magn Reson Med, 60:1396-1407';
addField.EstimationAlgorithm =  'src/Models_Functions/MTSATfun/MTSAT_exec.m';
provenance = Model.getProvenance('extra',addField);
savejson('',provenance,[pwd filesep 'output_rec' '.json']);
```

Reply:
sorry i was not clear: i meant those extra empty lines (there are a lot of those throughout the code)

Reply:
Oh, not at all. Removing them right away.

Reply:
849995a45c90a193d9ec8b538d000f9b5846bded